### PR TITLE
[SPARK-24981][Core] ShutdownHook timeout causes job to fail when succeeded when SparkContext stop() not called by user program

### PR DESCRIPTION
**Pull request by hthuynh2**
**Description**
The issue is described in [SPARK-24981](https://issues.apache.org/jira/browse/SPARK-24981). 

**How does this PR fix the issue?**
This PR catch the Exception that is thrown while the Sparkcontext.stop() is running (when it is called by the ShutdownHookManager).

**How was this patch tested?**
I manually tested it by adding delay (60s) inside the stop(). This make the shutdownHookManger interrupt the thread that is running stop(). The Interrupted Exception was catched and the job succeed.

### CODE REVIEW

**Comment by holdensmagicalunicorn** [comment]
@hthuynh2, thanks! I am a bot who has found some folks who might be able to help with the review:@vanzin, @rxin and @mateiz

**Comment by hthuynh2** [comment]
@tgravescs Can you please have a look at it? Thanks.

**Comment by tgravescs** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -571,7 +571,12 @@ class SparkContext(config: SparkConf) extends Logging {
     _shutdownHookRef = ShutdownHookManager.addShutdownHook(
       ShutdownHookManager.SPARK_CONTEXT_SHUTDOWN_PRIORITY) { () =>
       logInfo("Invoking stop() from shutdown hook")
-      stop()
+      try {
+        stop()
+      } catch {
+        case e: Throwable =>
+          logWarning("Ignoring Exception while stoping SparkContext. Exception: " + e)
```
Comment (felixcheung) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: `stoping` -> `stopping`

Comment (felixcheung) [reply core/src/main/scala/org/apache/spark/SparkContext.scala]: use this format 
`logWarning("", exception)`

Comment (hthuynh2) [reply core/src/main/scala/org/apache/spark/SparkContext.scala]: @felixcheung Thanks for the comments. I updated it.

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 2]
```diff
@@ -571,7 +571,12 @@ class SparkContext(config: SparkConf) extends Logging {
     _shutdownHookRef = ShutdownHookManager.addShutdownHook(
       ShutdownHookManager.SPARK_CONTEXT_SHUTDOWN_PRIORITY) { () =>
       logInfo("Invoking stop() from shutdown hook")
-      stop()
+      try {
+        stop()
+      } catch {
+        case e: Throwable =>
+          logWarning("Ignoring Exception while stopping SparkContext", e)
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: minor nit, could you add in "while stopping SparkContext from shutdownhook"

**Comment by hthuynh2** [comment]
@tgravescs I updated. Thanks.

**Comment by tgravescs** [comment]
+1 pending jenkins

**Comment by tgravescs** [comment]
merged, Thanks @hthuynh2

**Pull request closed without merging**